### PR TITLE
Nginx 0.21.x fix

### PR DESCRIPTION
**Pull request by orangedeng**
Update ingress controller version t 0.21.0 and add drone support

Related Issue https://github.com/rancher/rancher/issues/15163

### CODE REVIEW

File: Makefile_rancher [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	./.dapper -m bind trash
+
+trash-keep: .dapper
+	./.dapper -m bind trash -k
+
+deps: trash
+
+.DEFAULT_GOAL := ci
+
+.PHONY: $(TARGETS)
```
Comment (alena1108) [inline Makefile_rancher]: new line missing

File: Dockerfile.dapper [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ENV DAPPER_SOURCE /go/src/k8s.io/ingress-nginx/
+ENV DAPPER_OUTPUT ./bin ./dist
+ENV DAPPER_DOCKER_SOCKET true
+ENV DAPPER_ENV CROSS TAG
+ENV DAPPER_RUN_ARGS="--net host"
+ENV TRASH_CACHE ${DAPPER_SOURCE}/.trash-cache
+ENV HOME ${DAPPER_SOURCE}
+WORKDIR ${DAPPER_SOURCE}
+ENTRYPOINT ["./scripts/entry"]
+CMD ["ci"]
```
Comment (alena1108) [inline Dockerfile.dapper]: new line missing

File: scripts/envs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	QEMUARCH=aarch64
+fi
+if [ "$ARCH" = "ppc64le" ]; then
+	QEMUARCH=ppc64le
+	GOARCH=ppc64le
+	DUMB_ARCH=ppc64el
+fi
+if [ "$ARCH" = "s390x" ]; then
+	QEMUARCH=s390x
+fi
```
Comment (alena1108) [inline scripts/envs]: new line missing

File: scripts/package [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -e
+
+REPO=${REPO:-rancher}
+# SCRIPTPATH="$(cd "$(dirname "$0")"; pwd -P)"
```
Comment (alena1108) [inline scripts/package]: why do we have so many lines commented? can they be removed?

File: scripts/release [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+exec $(dirname $0)/ci
```
Comment (alena1108) [inline scripts/release]: new line missing

**Comment by alena1108** [comment]
I couldn't find any mapping to 0.21.0 nginx release in the pr; can you describe of what the pr is supposed to do? How would tagging to specific nginx version work

**Comment by orangedeng** [comment]
The base branch `nginx-0.21.x-fix` is even to the release 0.21.0 in [ingress-nginx](https://github.com/kubernetes/ingress-nginx/tree/nginx-0.21.0). So we only need to apply our changes in the head of the release.

**Pull request merged**